src/rest_utils.cpp

```diff
@@ -188,6 +193,253 @@ Status makeJsonFromPredictResponse(
+                if (!status.ok())
+                    return status;
+                for (auto& number : tensor.contents().uint_contents()) {
+                    tensor_data.PushBack(number, response.GetAllocator());
+                }
+            } else {
+                for (size_t i = 0; i < response_proto.raw_output_contents(tensor_it).size(); i += sizeof(uint32_t))
+                    tensor_data.PushBack(*(reinterpret_cast<const uint32_t*>(response_proto.raw_output_contents(tensor_it).data() + i)), response.GetAllocator());
+            }
+        }
+        else if(tensor.datatype() == "UINT16") {
+            if (seekDataInValField) {
+                auto status = checkValField(tensor.contents().uint_contents_size(), expectedElementsNumber);
+                if (!status.ok())
+                    return status;
+                for (auto& number : tensor.contents().uint_contents()) {
+                    tensor_data.PushBack(number, response.GetAllocator());
+                }
+            } else {
+                for (size_t i = 0; i < response_proto.raw_output_contents(tensor_it).size(); i += sizeof(uint16_t))
+                    tensor_data.PushBack(*(reinterpret_cast<const uint16_t*>(response_proto.raw_output_contents(tensor_it).data() + i)), response.GetAllocator());
+            }
+        }
+        else if(tensor.datatype() == "UINT8") {
+            if (seekDataInValField) {
+                auto status = checkValField(tensor.contents().uint_contents_size(), expectedElementsNumber);
+                if (!status.ok())
+                    return status;
+                for (auto& number : tensor.contents().uint_contents()) {
+                    tensor_data.PushBack(number, response.GetAllocator());
+                }
+            } else {
+                for (size_t i = 0; i < response_proto.raw_output_contents(tensor_it).size(); i += sizeof(uint8_t))
+                    tensor_data.PushBack(*(reinterpret_cast<const uint8_t*>(response_proto.raw_output_contents(tensor_it).data() + i)), response.GetAllocator());
+            }
+        }
+        else if(tensor.datatype() == "FP64") {
+            if (seekDataInValField) {
+                auto status = checkValField(tensor.contents().fp64_contents_size(), expectedElementsNumber);
+                if (!status.ok())
+                    return status;
+                for (auto& number : tensor.contents().fp64_contents()) {
+                    tensor_data.PushBack(number, response.GetAllocator());
+                }
+            } else {
+                for (size_t i = 0; i < response_proto.raw_output_contents(tensor_it).size(); i += sizeof(double))
+                    tensor_data.PushBack(*(reinterpret_cast<const double*>(response_proto.raw_output_contents(tensor_it).data() + i)), response.GetAllocator());
+            }
+        }
+        else{
+            return StatusCode::REST_UNSUPPORTED_PRECISION;
+        }
```

Comment:
1. That's pretty long block, maybe it could be extracted to a function?
2. The elements of that block look quite similar, maybe be could have a function for it as well?